azurerm/provider_test.go

```diff
@@ -48,6 +51,17 @@ func testAccPreCheck(t *testing.T) {
 			t.Fatalf("`%s` must be set for acceptance tests!", variable)
 		}
 	}
+
+	// If "TF_PARA" is set to some non-empty value, all the parallel test cases
+	// will be run in parallel. It requires that the test cases are all
+	// independent ones without any shared resource before turn it on.
+	// example usage:
+	//  TF_ACC=1 TF_PARA=1 go test [TEST] [TESTARGS] -v -parallel=n
+	// To run specified cases in parallel, please refer to below PR:
+	// 	https://github.com/hashicorp/terraform/pull/16807
+	if os.Getenv(ParaTestEnvVar) != "" {
+		t.Parallel()
+	}
```

Comment:
I agree with you, that's why I added similar flag in this [PR](https://github.com/hashicorp/terraform/pull/16807), but since resource's test suite is widely used by many different providers, I set its default value be false since I don't want to break any legacy code. The current test cases in azurerm provider have very strong dependencies in resource.Test and resource.TestCase, if we want to introduce another different flag(RunInParallel) on this provider only, we probably need to introduce a middle layer to handle this.

For the acc tests in "resource_arm_network_watcher_test.go", I am curious that how you handle it today in TeamCity CI coz it's also parallel running environment. Can we use subtests to group the current test cases for network watcher ? By this way, we can still keep it running with other tests in parallel, but its subtests run in sequentially.